### PR TITLE
5 view individual article

### PR DESCRIPTION
**Pull request by im-ilyasmohamed**
Set up routes, refactored some things, and created an article page with components that render with a useState, displaying relevant components for the  individual article page

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,17 +1,79 @@
...
+import axios from "axios";
+import { useEffect, useState } from "react";
+import { Routes, Route } from "react-router-dom";
+import { Link } from "react-router-dom";
+// Creating a custom axios instance
+//
+const apiClient = axios.create({
+  baseURL: "https://my-backend-project-bc-news.onrender.com",
+  timeout: 1000,
+});
```
Comment (nc-christian-dutton) [inline src/App.jsx:16]: You should put this in a separate file for all your api calls!

File: src/App.jsx [hunk 2]
```diff
@@ -1,17 +1,79 @@
...
+        // handle success
 
+        setArticles(response.data.allArticles);
+        setLoading(false); // so can render return statement below
+      })
+      .catch((error) => {
+        // handle error
+        setLoading(false); // stops the loading state, if cant get the data
+      });
+  }, []); // important to have a dependency array even if blank, this ensures that the resource is not loaded on every render, i.e. only set on first mount and not rerendered
```
Comment (nc-christian-dutton) [inline src/App.jsx:44]: All of the above should not be in app.jsx - it should be in articlesList. You should clear up these comments too at some point, leave them for now if they are helping?

File: src/App.jsx [hunk 3]
```diff
@@ -1,17 +1,79 @@
...
+          element={
+            <ArticlesList
+              articles={articles}
+              loading={loading}
+              setCurrentArticleIndividual={setCurrentArticleIndividual}
+            />
+          }
+        />
+        <Route
+          path="/article"
```
Comment (nc-christian-dutton) [inline src/App.jsx:65]: This should go to article/:article_id - that means you could use useParams in articleIndividual to grab the article_id.

File: src/components/ArticleIndividual/ArticleIndividual.jsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+function ArticleIndividual({ loading, currentArticleIndividual }) {
+  function helperFunction() {
+    console.log(currentArticleIndividual, "individual article page");
+  }
+  helperFunction();
+
+  // could refactor to add loading states
```
Comment (nc-christian-dutton) [inline src/components/ArticleIndividual/ArticleIndividual.jsx:7]: You want an api call here to /api/articles/:article_id - to get the individual article. You can use useParams to grab the article_id from the URL.

File: src/components/ArticlesList/ArticleCard.jsx [hunk 5]
```diff
@@ -10,7 +15,11 @@ function ArticleCard({ article_item }) {
         <h3>{article_item.title}</h3>
         <h4>{article_item.topic}</h4>
         <p className="p-text-ellipsis">{article_item.body}</p>
-        <button type="button">Read Article</button>
+        <Link to="/article">
```
Comment (nc-christian-dutton) [inline src/components/ArticlesList/ArticleCard.jsx:18]: Make the link here to /article/${article.article_id} - then match that link in your routes (I've suggested above)

**Comment by nc-christian-dutton** [comment]
You can close PR's straight away you don't need to wait for us! Once you send it, close it and merge

File: src/App.jsx (hunk 1 above)
Comment (im-ilyasmohamed) [reply src/App.jsx:16]: done

File: src/App.jsx (hunk 2 above)
Comment (im-ilyasmohamed) [reply src/App.jsx:44]: done

File: src/App.jsx (hunk 3 above)
Comment (im-ilyasmohamed) [reply src/App.jsx:65]: done

**Pull request merged**